File: BookLocal.Data/Models/ServiceVariant.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookLocal.Data.Models
{
    public class ServiceVariant
    {
        [Key]
        public int ServiceVariantId { get; set; }

        [Required]
        public int ServiceId { get; set; }
        [ForeignKey("ServiceId")]
        public virtual Service Service { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        [Column(TypeName = "decimal(10, 2)")]
        public decimal Price { get; set; }

        [Required]
        public int DurationMinutes { get; set; }

        [Required]
        public int CleanupTimeMinutes { get; set; } = 0;

        public bool IsDefault { get; set; } = false;
        public bool IsActive { get; set; } = true;

        public virtual ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
        public virtual ICollection<UserFavoriteService> UserFavoriteServices { get; set; } = new List<UserFavoriteService>();
    }
}

File: booklocal_mobile/lib/core/models/service_variant.dart
class ServiceVariant {
  final int serviceVariantId;
  final String name;
  final double price;
  final int durationMinutes;
  final int cleanupTimeMinutes;
  final bool isDefault;
  final bool isActive;
  final int favoritesCount;

  ServiceVariant({
    required this.serviceVariantId,
    required this.name,
    required this.price,
    required this.durationMinutes,
    required this.cleanupTimeMinutes,
    required this.isDefault,
    required this.isActive,
    required this.favoritesCount,
  });

  factory ServiceVariant.fromJson(Map<String, dynamic> json) {
    return ServiceVariant(
      serviceVariantId: json['serviceVariantId'] ?? 0,
      name: json['name'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      durationMinutes: json['durationMinutes'] ?? 0,
      cleanupTimeMinutes: json['cleanupTimeMinutes'] ?? 0,
      isDefault: json['isDefault'] ?? false,
      isActive: json['isActive'] ?? true,
      favoritesCount: json['favoritesCount'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'serviceVariantId': serviceVariantId,
      'name': name,
      'price': price,
      'durationMinutes': durationMinutes,
      'cleanupTimeMinutes': cleanupTimeMinutes,
      'isDefault': isDefault,
      'isActive': isActive,
      'favoritesCount': favoritesCount,
    };
  }
}
